Content.Client/Disposal/Visualizers/DisposalUnitVisualizerSystem.cs

```diff
@@ -1,104 +1,94 @@
     {
+        [Dependency] private readonly SharedAudioSystem SoundSystem = default!;
         private const string AnimationKey = "disposal_unit_animation";
 
-        [DataField("state_anchored", required: true)]
-        private string? _stateAnchored;
-
-        [DataField("state_unanchored", required: true)]
-        private string? _stateUnAnchored;
-
-        [DataField("state_charging", required: true)]
-        private string? _stateCharging;
-
-        [DataField("overlay_charging", required: true)]
-        private string? _overlayCharging;
-
-        [DataField("overlay_ready", required: true)]
-        private string? _overlayReady;
-
-        [DataField("overlay_full", required: true)]
-        private string? _overlayFull;
-
-        [DataField("overlay_engaged", required: true)]
-        private string? _overlayEngaged;
-
-        [DataField("state_flush", required: true)]
-        private string? _stateFlush;
-
-        [DataField("flush_sound", required: true)]
-        private SoundSpecifier _flushSound = default!;
-
-        [DataField("flush_time", required: true)]
-        private float _flushTime;
+        public override void Initialize()
+        {
+            base.Initialize();
 
-        private Animation _flushAnimation = default!;
+            SubscribeLocalEvent<DisposalUnitComponent, ComponentInit>(OnComponentInit);
+        }
 
-        void ISerializationHooks.AfterDeserialization()
+        private void OnComponentInit(EntityUid uid, DisposalUnitComponent disposalUnit, ComponentInit args)
         {
-            _flushAnimation = new Animation {Length = TimeSpan.FromSeconds(_flushTime)};
+            disposalUnit.FlushAnimation = new Animation {
+                Length = TimeSpan.FromSeconds(disposalUnit.FlushTime)
+            };
 
             var flick = new AnimationTrackSpriteFlick();
-            _flushAnimation.AnimationTracks.Add(flick);
+            disposalUnit.FlushAnimation.AnimationTracks.Add(flick);
             flick.LayerKey = DisposalUnitVisualLayers.Base;
-            flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(_stateFlush, 0));
+            flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(disposalUnit.StateFlush, 0));
 
             var sound = new AnimationTrackPlaySound();
-            _flushAnimation.AnimationTracks.Add(sound);
+            disposalUnit.FlushAnimation.AnimationTracks.Add(sound);
+
+            sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(SoundSystem.GetSound(disposalUnit.FlushSound), 0));
+
+            var entities = IoCManager.Resolve<IEntityManager>();
+            entities.EnsureComponent<AnimationPlayerComponent>(uid);
+            var appearance = entities.EnsureComponent<ClientAppearanceComponent>(uid);
 
-            sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(_flushSound.GetSound(), 0));
+            if (entities.TryGetComponent(uid, out ISpriteComponent? sprite) && sprite != null)
```

Comment:
EntitySystem has proxy methods for this, you dont have to resolve, just use `EnsureComp<>(),` and `TryComp` also why are you ensuring appearance component?

Reply:
Thanks, didn't even see that property on EntitySystem. Must have missed that AppearanceComponent - leftover from the old code.